pkg/restapi/issuer/operation/operations.go

```diff
@@ -145,8 +145,17 @@ func (o *Operation) getIssuerProfileHandler(rw http.ResponseWriter, req *http.Re
 	commhttp.WriteResponse(rw, profile)
 }
 
-func (o *Operation) walletConnect(w http.ResponseWriter, r *http.Request) {
-	http.Redirect(w, r, o.uiEndpoint, http.StatusFound)
+func (o *Operation) walletConnect(rw http.ResponseWriter, req *http.Request) {
```

Comment:
Can you describe the flow here?

If I understand correctly, the user is on the Issuer's page and they click on a button. This button redirects the user to this endpoint, right? The redirect will contain the issuer's profileID, which I guess may be hardcoded into that button.

This method then redirects the user to the frontend. The frontend has a button "Connect Wallet". It makes a request to `/issuer/didcomm/invitation` without parameters.

Questions:

* Won't it be a more streamlined flow if the issuer's page redirected the user to the issuer adapter frontend directly? The redirect already contains the profileID.
* There is no correlation that I can see with the request to `/issuer/didcomm/invitation`. Is that on your TODO?

Reply:
Redirecting to backend helps to to setup the session details (wip) and validations. Based on that, we can show failure message or proceed with the flow. Rather than directly showing UI and then validating the details.
Yes, I'm analyzing that.